src/main/java/com/example/demo/controller/ProductController.java

```diff
@@ -61,4 +69,20 @@ public ResponseEntity<Void> deleteProduct(@PathVariable Long id) {
         }
         return ResponseEntity.notFound().build();
     }
+
+    private ProductDto mapToProductDto(Product product) {
+        ProductDto productDto = new ProductDto();
+        productDto.setId(product.getId());
+        productDto.setName(product.getName());
+        productDto.setPrice(product.getPrice());
+        return productDto;
+    }
+
+    private Product mapToProduct(ProductDto productDto) {
+        Product product = new Product();
+        product.setId(productDto.getId());
+        product.setName(productDto.getName());
+        product.setPrice(productDto.getPrice());
+        return product;
+    }
```

Comment:
This way of mapping works well, but doesn't scale. If you were to add another field in the Product class, you'd need to remember to come back here and reflect this here.  Use the modelMapper you've setup in a config class before https://www.baeldung.com/java-modelmapper

Reply:
I have decided not to use mapToProduct in the controllers anymore. I have made changes in the controllers to use ModelMapper for mapping DTO objects to model objects and vice versa. We now use modelMapper.map() for conversion between DTO objects and models.